testhelpers/org.eclipse.n4js.ui.tests.helper/src/org/eclipse/n4js/tests/util/EclipseUIUtils.java

```diff
@@ -73,14 +74,20 @@ public static IWorkbenchPage getActivePage() {
 	 */
 	public static IViewPart showView(final String id) {
 		checkNotNull(id, "Provided view ID was null.");
-		try {
-			IWorkbenchPage activePage = getActivePage();
-			return activePage.showView(id);
-		} catch (final PartInitException e) {
-			final String message = "Error occurred while initializing view with ID: '" + id + "'.";
-			LOGGER.error(message, e);
-			throw new RuntimeException(message, e);
-		}
+
+		final AtomicReference<IViewPart> result = new AtomicReference<>();
+		final Display display = PlatformUI.getWorkbench().getDisplay();
+		display.syncExec(() -> {
```

Comment:
Normally I'd say a simple utility method should not do this, but other methods in this class also use syncExec(), so seems in line. But there are now some inconsistencies in this class w.r.t. UI/non-UI thread. Let's discuss...

Reply:
Thanks for the discussion. We decided to leave it like that since it is test code only, and to keep it in mind for a future clean-up/refactoring. We made a note in a user story to ask SZ.